test/e2e/csv_e2e_test.go

```diff
@@ -3465,44 +3376,47 @@ func waitForDeployment(c operatorclient.ClientInterface, name string) error {
 	})
 }
 
-func waitForDeploymentToDelete(t GinkgoTInterface, c operatorclient.ClientInterface, name string) error {
-	return wait.Poll(pollInterval, pollDuration, func() (bool, error) {
-		t.Logf("waiting for deployment %s to delete", name)
+func waitForDeploymentToDelete(c operatorclient.ClientInterface, name string) error {
+	var err error
+
+	Eventually(func() (bool, error) {
+		ctx.Ctx().Logf("waiting for deployment %s to delete", name)
 		_, err := c.GetDeployment(testNamespace, name)
 		if errors.IsNotFound(err) {
-			t.Logf("deleted %s", name)
+			ctx.Ctx().Logf("deleted %s", name)
 			return true, nil
 		}
 		if err != nil {
-			t.Logf("err trying to delete %s: %s", name, err)
+			ctx.Ctx().Logf("err trying to delete %s: %s", name, err)
 			return false, err
 		}
 		return false, nil
-	})
+	}).Should(BeTrue())
+	return err
 }
 
-func waitForCSVToDelete(t GinkgoTInterface, c versioned.Interface, name string) error {
+func waitForCSVToDelete(c versioned.Interface, name string) error {
```

Comment:
With this change, won't the return value always be nil? `err` is shadowed in the closure, so the `err` declared on L3399 is never assigned. Besides, the test will end immediately if the Eventually fails, so no return value ever be read.

I think the idiomatic way to adapt this is to replace calls to `waitForCSVToDelete` with Eventually assertions. If this is used all over the place, and the duplication is unbearable, then it could make sense to define a function `csvExists` and write `Eventually(csvExists).Should(BeFalse())`.

Reply:
Added `func csvExists`